src/main/java/at/hannibal2/skyhanni/features/rift/everywhere/PunchcardHighlight.kt

```diff
@@ -0,0 +1,235 @@
+            ) {
+                reloadColors()
+                lastRiftServer = HypixelData.server
+                playerList.clear()
+            }
+        }
+    }
+
+    private fun colorPlayer(entity: EntityLivingBase) {
+        val color = config.color.get().toChromaColor()
+        val alpha = when (color.alpha) {
+            0 -> 0
+            255 -> 1
+            else -> 255 - color.alpha
+        }
+        RenderLivingEntityHelper.setEntityColor(entity, color.withAlpha(alpha)) { IslandType.THE_RIFT.isInIsland() }
+    }
+
+    private fun removePlayerColor(entity: EntityLivingBase) {
+        RenderLivingEntityHelper.removeEntityColor(entity)
+    }
+
+    fun clearList() {
+        playerList.clear()
+        playerQueue.clear()
+        if (config.reverse.get()) {
+            MobData.players.forEach {
+                colorPlayer(it.baseEntity)
+            }
+        } else {
+            MobData.players.forEach {
+                removePlayerColor(it.baseEntity)
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onPunch(event: EntityClickEvent) {
+        if (!RiftAPI.inRift()) return
+        val entity = event.clickedEntity
+        if (entity !is AbstractClientPlayer) return
+        if (entity.isNPC()) return
+        val name = entity.name
+        if (name in playerList || name in playerQueue) return
+        playerQueue.add(name)
+        listening = true
+        DelayedRun.runDelayed(1.seconds) {
+            if (name in playerQueue) playerQueue.remove(name)
+            listening = false
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!IslandType.THE_RIFT.isInIsland()) return
+        if (!listening) return
+        if (playerQueue.isEmpty()) return
+        val message = event.message
+        val queuedName = playerQueue[0]
+        punchedPattern.matchMatcher(message) {
+            val name = group("name")
+            if (queuedName == name) {
+                addPunch(name)
+            } else ErrorManager.logErrorStateWithData(
+                "Error finding punched player", "queuedName and capturedName were different",
+                "queuedName" to queuedName,
+                "capturedName" to name,
+                noStackTrace = true,
+                betaOnly = true,
+            )
+            return
+        }
+        if (limitPattern.matches(message) || repeatPattern.matches(message)) addPunch(queuedName)
+    }
+
+    private fun addPunch(playerName: String) {
+        playerList.add(playerName)
+        playerQueue.remove(playerName)
+        val player = MobData.players.firstOrNull { it.name == playerName } ?: return
+        if (!config.reverse.get()) removePlayerColor(player.baseEntity)
+        else colorPlayer(player.baseEntity)
+        display = drawDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderUI(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!config.gui.get()) return
```

Comment:
first check for the server, then the config. this could cause a confusing crash otherwise.